docs/plots/backends.rst

```diff
@@ -0,0 +1,389 @@
+annotation might differ, but they will convey the same information. For example,
+"a blue dotted line" will generally appear blue and dotted, even though the
+shade of blue or the size of the dots might differ between plotting backends.
+
+In some cases, a plotting backend might not support all Sherpa plot options (for
+example, a plotting backend might not have a command to change the line style).
+In those rare cases, a setting might be ignored; but a setting from Sherpa's
+list of backend independent values will never raise an error.
+
+The names of plotting options and values that all Sherpa plotting backends
+accept are chosen to match common :term:`matplotlib` settings which are familiar
+to many scientific Python users, and to maintain backwards compatibility with
+previous version of Sherpa. They offer limited choices, but those are sufficient
+for most plots.
+
+The following settings are accepted for all Sherpa plotting backends:
+
+Colors
+------
+
+Colors are used for the color of lines, symbols, error bars etc.
+
+- ``'b'`` (blue)
+- ``'r'`` (red)
+- ``'g'`` (green)
+- ``'k'`` (black)
+- ``'w'`` (white),
+- ``'c'`` (cyan)
+- ``'y'`` (yellow)
+- ``'m``` (magenta)
+- `None` (plotting backend default)
+
+Line styles
+------------
+
+Because of legacy from both Chips and :term:`matplotlib` backends, line styles
+can be specified in more than one form:
+
+- ``'noline'`` or  ``''`` (empty string) or ``'None'`` (as a string)
+  means that no line shall be plotted.
+- ``'solid'`` or ``'-'``  for solid lines,
+- `None`  for the default style - usually a solid line, too,
+- ``'dot'`` or ``':'`` for dotted lines,
+- ``'dash'`` or ``'--'`` (dashed) for dashed lines, and
+- ``'dotdash'`` or ``'-.'`` for a dot-dashed line.
+
+Markers
+--------
+- ``"None"`` as a string or ``""`` (empty string) means that no marker will be
+  shown,
+- ``"."`` shows dots,
+-  ``"o"`` shows circles (filled or unfilled depending on the backend),
+- ``"+"`` shows plus signs,
+- ``"s"`` shows squares.
+
+Additional backend-specific settings
+====================================
+
+Most plotting backends accept more than just the backend-independent options
+listed above. For example, :term:`matplotlib` allows `many different ways to
+specify colors to cover the entire RGB range
+<https://matplotlib.org/stable/tutorials/colors/colors.html>`_, such as
+``color=(.3, .4, .5, .2)`` or ``color='xkcd:eggshell'``. The Sherpa plotting
+methods will pass any value to the underlying backend::
+
+  >>> from sherpa.data import Data1D
+  >>> from sherpa.plot import DataPlot
+  >>> d = Data1D('example data', [1, 2, 3], [3, 2, 5])
+  >>> dplot = DataPlot()
+  >>> dplot.prepare(d)
+  >>> dplot.plot(markerfacecolor='xkcd:eggshell')
+
+will succeed with the `~sherpa.plot.pylab_backend.PylabBackend`
+(activated with ``sherpa.plot.set_backend("pylab")``), but raise an
+error if the active backend is `~sherpa.plot.bokeh_backend.BokehBackend`. In
```

Comment:
Unfortunately, at the moment there is no `sherpa.plot.bokeh_backend` module. In fact, the remaining text in this paragraph assumes that we have a bokeh backend.

I am willing to let this pass now (as re-writing this would be a lot of wasted work once we get a bokeh backend) but we should check again just before the 4.16.0 code freeze.

Reply:
[also hoping we get a bokeh backend soon so then we don't need to worry]

Reply:
I have a pretty complete draft. The reason I've not push that yet is that it's a pain to deal with rebases of this branch, which it's build on top of. However, with @DougBurke saying it's close, I'll push the bokeh backend shortly. I do want to get both in.